playbooks/aws/openshift-cluster/tasks/launch_instances.yml

```diff
@@ -34,6 +35,32 @@
     ec2_assign_public_ip: "{{ lookup('env', 'ec2_assign_public_ip')
                     | default(deployment_vars[deployment_type].assign_public_ip, true) }}"
   when: ec2_assign_public_ip is not defined
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_master_instance_type }}"
+  when: host_type == "master"
+- set_fact:
+    ec2_security_groups: "{{ ec2_master_security_groups }}"
+  when: host_type == "master"
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_infra_instance_type }}"
+  when: sub_host_type == "infra"
+- set_fact:
+    ec2_security_groups: "{{ ec2_infra_security_groups }}"
+  when: sub_host_type == "infra"
+
+- set_fact:
+    ec2_instance_type: "{{ ec2_node_instance_type }}"
+  when: sub_host_type == "node"
```

Comment:
If host_type is master it loads the master configs, other wise it filters on sub_host_type, it is a single task that handles master, etcd, and node creations.